### PR TITLE
feat(library): Introduce Nginx 1.25 as library

### PR DESCRIPTION
**Pull request by razvand**
Introduce Nginx 1.25 image (ELF Loader) for bincompat runs.

Add `Kraftfile` to build for QMEU and Firecracker on x86_64 using initrd for volume support. The current build uses embedded initrd to build a single image consisting of both the kernel and the initial ramdisk.

Add GitHub workflow to build and deploy the image. Add success badge in `README.md`.

### CODE REVIEW

**Review by StefanJum — APPROVED** [review]
All good with the local build.

File: library/nginx/1.25/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+COPY --from=build /lib/x86_64-linux-gnu/libcrypto.so.3 /lib/x86_64-linux-gnu/libcrypto.so.3
+COPY --from=build /lib/x86_64-linux-gnu/libz.so.1 /lib/x86_64-linux-gnu/libz.so.1
+COPY --from=build /lib/x86_64-linux-gnu/libc.so.6 /lib/x86_64-linux-gnu/libc.so.6
+COPY --from=build /lib64/ld-linux-x86-64.so.2 /lib64/ld-linux-x86-64.so.2
+COPY --from=build /etc/ld.so.cache /etc/ld.so.cache
+
+# Custom configuration files, including using a single process for Nginx
+COPY ./conf/nginx.conf /etc/nginx/nginx.conf
+COPY ./conf/unikraft.local.crt /etc/nginx/unikraft.local.crt
+COPY ./conf/unikraft.local.key /etc/nginx/unikraft.local.key
```
Comment (nderjung) [inline library/nginx/1.25/Dockerfile:34]: I think we should not include faux certificates, please can you remove? a). they are used for demo purposes, and demo purposes can use HTTP; and, b). they cannot be removed unless the user re-builds the image, potential user who uses this out-of-the-box could be at risk if they do choose to use this certificate.

File: library/nginx/1.25/conf/nginx.conf [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  keepalive_requests    10000;
+  send_timeout          10s;
+
+  server {
+    listen              80;
+    listen              443 ssl;
+
+    ssl_protocols       TLSv1.2;
+    ssl_certificate     unikraft.local.crt;
+    ssl_certificate_key unikraft.local.key;
```
Comment (nderjung) [inline library/nginx/1.25/conf/nginx.conf:31]: Can you pls remove ssl related functionality (ln 27-31).

**Comment by felipehuici** [comment]
Reviewed-by: Felipe Huici <felipe@unikraft.io>
Approved-by: Felipe Huici <felipe@unikraft.io>

**Pull request merged**